src/test/java/MzMLImportTest.java

```diff
@@ -0,0 +1,145 @@
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_b.mzXML").getFile()), //ind 1
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_a_invalid_header.mzML").getFile()), //ind 2
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_b_invalid_header.mzXML").getFile()) //ind 3
+        };
+
+        AllSpectralDataImportParameters paramDataImport = new AllSpectralDataImportParameters();
+        paramDataImport.setParameter(AllSpectralDataImportParameters.fileNames, files);
+        paramDataImport.setParameter(SpectralLibraryImportParameters.dataBaseFiles, new File[0]);
+        paramDataImport.setParameter(AllSpectralDataImportParameters.advancedImport, false);
+
+        logger.info("Testing data import of mzML and mzXML without advanced parameters");
+        TaskResult finished = MZmineTestUtil.callModuleWithTimeout(30,
+                AllSpectralDataImportModule.class, paramDataImport);
+
+        // should have finished by now
+        assertEquals(TaskResult.FINISHED, finished, () -> switch (finished) {
+            case TIMEOUT -> "Timeout during data import. Not finished in time.";
+            case ERROR -> "Error during data import.";
+            case FINISHED -> "";
+        });
+
+        assertEquals(4, project.getDataFiles().length);
+
+        //todo check all scans and mass lists
+        for (RawDataFile raw : project.getCurrentRawDataFiles()) {
+            for (Scan scan : raw.getScans()) {
+                assertNotNull(scan);
+//                assertNotNull(scan.getMassList());
+            }
+        }
+
+        //check if number of scans matches in files with/without 3-byte symbols
+        RawDataFile rawA = project.getCurrentRawDataFiles().get(2);
+        RawDataFile rawB = project.getCurrentRawDataFiles().get(1);
+        RawDataFile rawAInvalid = project.getCurrentRawDataFiles().get(3);
+        RawDataFile rawBInvalid = project.getCurrentRawDataFiles().get(0);
+
+        assertEquals(521, rawA.getNumOfScans());
+        assertEquals(521, rawB.getNumOfScans());
+
+        assertEquals(rawA.getNumOfScans(),
+                rawAInvalid.getNumOfScans());
+        assertEquals(rawB.getNumOfScans(),
+                rawBInvalid.getNumOfScans());
+
+        //check number of data points
+        //todo centroid??
+        assertEquals(2400, rawA.getMaxRawDataPoints());
+        assertEquals(rawA.getMaxRawDataPoints(), rawAInvalid.getMaxRawDataPoints());
+        assertEquals(rawA.getMaxCentroidDataPoints(), rawAInvalid.getMaxCentroidDataPoints());
+
+        assertEquals(2410, rawB.getMaxRawDataPoints());
+        assertEquals(rawB.getMaxRawDataPoints(), rawBInvalid.getMaxRawDataPoints());
+        assertEquals(rawB.getMaxCentroidDataPoints(), rawBInvalid.getMaxCentroidDataPoints());
+
+        //check MS1, MS2 scans
+        assertEquals(87, rawA.getScanNumbers(1).size());
```

Comment:
maybe do the same for some later scan as well.
Also read some metadata maybe

Reply:
a later scan? here I just get scan numbers for ms levels 1 and 2